connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -307,15 +307,11 @@ private void processConnectorConfigUpdates(Set<String> connectorConfigUpdates) {
     }
 
     private void processTargetStateChanges(Set<String> connectorTargetStateChanges) {
-        if (!connectorTargetStateChanges.isEmpty()) {
-            for (String connector : connectorTargetStateChanges) {
-                if (worker.connectorNames().contains(connector)) {
-                    TargetState targetState = configState.targetState(connector);
-                    worker.setTargetState(connector, targetState);
-                    if (targetState == TargetState.STARTED)
-                        reconfigureConnectorTasksWithRetry(connector);
-                }
-            }
+        for (String connector : connectorTargetStateChanges) {
+            TargetState targetState = configState.targetState(connector);
+            worker.setTargetState(connector, targetState);
+            if (targetState == TargetState.STARTED)
+                reconfigureConnectorTasksWithRetry(connector);
```

Comment:
I think this is going to end up throwing an exception if it executes on a worker that isn't running the connector. `reconfigureConnectorTasksWithRetry` needs to only be executed on the worker actually running the connector since the first check it does to check if the connector is running throws an exception if the connector doesn't exist.

I think you want to call `worker.setTargetState` without the check, but still maintain the check around this call.

The current tests miss this for two reasons. First, they exercise pausing but not resuming. Second, the result is not a crash, but log noise and rescheduling the next attempt to reconfigure the connector.